src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -350,7 +349,7 @@ private SocketError FinishOperationAccept(SocketAddress remoteSocketAddress)
 
             Socket acceptedSocket = _currentSocket!.CreateAcceptSocket(
                 SocketPal.CreateSocket(_acceptedFileDescriptor),
-                _currentSocket._rightEndPoint!.Create(remoteSocketAddress));
+                remoteSocketAddress.Size > 0 ? _currentSocket._rightEndPoint!.Create(remoteSocketAddress) : null);
```

Comment:
How do we end up with a 0 Size? Can we add a test?

Reply:
I was trying to reproduce it locally .... but I failed. 

The discussion was here: https://github.com/dotnet/runtime/issues/102663#issuecomment-2321533894
Verification of #108026 is still pending but looks promising IMHO.

Reply:
BTW I replaced the backport form with original description to me kit more clear.

Reply:
IMO it would be valuable to add a comment describing the reason for the conditional check.

Reply:
comment with explanation added.